flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/sink/KafkaWriterITCase.java

```diff
@@ -491,6 +498,9 @@ private class DummyRecordSerializer implements KafkaRecordSerializationSchema<In
         @Override
         public ProducerRecord<byte[], byte[]> serialize(
                 Integer element, KafkaSinkContext context, Long timestamp) {
+            if (element == null) {
+                return null;
+            }
             return new ProducerRecord<>(topic, ByteBuffer.allocate(4).putInt(element).array());
         }
     }
```

Comment:
Isn't this update of the current `testIncreasingRecordBasedCounters` case enough for testing the change @MartijnVisser?